scico/functional/_norm.py

```diff
@@ -477,3 +479,110 @@ def prox(
         svdU, svdS, svdV = snp.linalg.svd(v, full_matrices=False)
         svdS = snp.maximum(0, svdS - lam)
         return svdU @ snp.diag(svdS) @ svdV
+
+
+class TV2DNorm(Functional):
+    r"""The anisotropic total variation (TV) norm.
+
+    For a :math:`M \times N` matrix, :math:`\mb{A}`, by default,
+
+    .. math::
+           \norm{\mb{A}}_{\text{TV}} = \sum_{n=1}^N \sum_{m=1}^M
+           \abs{\nabla{A}_{m,n}} \;.
+
+    The proximal operator of this norm is currently only defined for 2
+    dimensional data.
+
+    For `BlockArray` inputs, the TV norm follows the reduction rules
```

Comment:
Since there doesn't seems to be an obvious way to generalize to `BlockArray` inputs, this should be removed.

Reply:
Sure